src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php

```diff
@@ -62,7 +64,8 @@ public function reverseTransform(mixed $value): int|float|null
     {
         $value = parent::reverseTransform($value);
         if (null !== $value && 1 !== $this->divisor) {
-            $value = (float) (string) ($value * $this->divisor);
+            $value = (string) ($value * $this->divisor);
+            $value = 'integer' === $this->modelType ? (int) $value : (float) $value;
```

Comment:
If the `model_type = integer` only works when `divisor` isn't 1, I'd raise a logic exception (for better DX) if the given options don't match this condition? You could add this in an option normalizer, similar to the `grouping` normalizer.

Reply:
I implemented the suggested change